lib/bundler/rubygems_integration.rb

```diff
@@ -67,6 +67,9 @@ def configuration
       Bundler.ui.error "#{e.class}: #{e.message}"
       Bundler.ui.trace e
       raise Gem::SystemExitException
+    rescue Psych::SyntaxError => e
+      raise YAMLSyntaxError.new(e, "Your RubyGems configuration file (usually located at ~/.gemrc) " \
```

Comment:
`Gem.configuration` is the method that actually raises the `Psych::SyntaxError`, so we can't really do `Gem.configuration.config_file_name`.